src/api/commands/profile/ProfileUpdateCommand.ts

```diff
@@ -40,6 +43,31 @@ export class ProfileUpdateCommand extends BaseCommand implements RpcCommandInter
         });
     }
 
+    public async validate(data: RpcRequest): Promise<RpcRequest> {
+        if (data.params.length < 1) {
+            throw new MissingParamException('profileId');
+        } else if (data.params.length < 2) {
+            throw new MissingParamException('newProfileName');
+        }
+
+        const profileId = data.params[0];
+        if (typeof profileId !== 'number') {
+            throw new InvalidParamException(profileId, 'number');
+        }
+
+        const newProfileName = data.params[1];
+        if (typeof newProfileName !== 'string') {
+            throw new InvalidParamException(newProfileName, 'string');
+        }
+
+        const profile = this.profileService.findOne(profileId);
```

Comment:
no need to call findOne here either, update will call that and throw NotFoundException when needed